### ISSUE TITLE
Jackson Binding Exception while checking for namespace

### ISSUE THREAD

**Issue by denizdurmus** [nemtech/nem2-sdk-java#2]
Based on the example here https://github.com/nemtech/nem2-docs/blob/master/source/resources/examples/java/src/test/java/nem2/guides/examples/namespace/CheckingNamespaceExistence.java I tried to see what happens if I sent a non-existing namespace. 

I have modified the code on the link slightly:

```java
try{
            final NamespaceId namespaceId = new NamespaceId("fooz");
            final NamespaceHttp namespaceHttp = new NamespaceHttp("http://localhost:3000");
            final NamespaceInfo namespaceInfo = namespaceHttp.getNamespace(namespaceId).toFuture().get();
            System.out.println(new Gson().toJson(namespaceInfo));
        }catch(MalformedURLException|InterruptedException|ExecutionException e){
            e.printStackTrace();
        }
```
```
java.util.concurrent.ExecutionException: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "code" (class io.nem.sdk.infrastructure.NamespaceInfoDTO), not marked as ignorable (2 known properties: "namespace", "meta"])
 at [Source: (String)"{"code":"InvalidArgument","message":"namespaceId has an invalid format"}"; line: 1, column: 10] (through reference chain: io.nem.sdk.infrastructure.NamespaceInfoDTO["code"])
	at io.reactivex.internal.observers.FutureObserver.get(FutureObserver.java:85)
	at com.xyz.cryptoplayground.App.main(App.java:58)
Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "code" (class io.nem.sdk.infrastructure.NamespaceInfoDTO), not marked as ignorable (2 known properties: "namespace", "meta"])
 at [Source: (String)"{"code":"InvalidArgument","message":"namespaceId has an invalid format"}"; line: 1, column: 10] (through reference chain: io.nem.sdk.infrastructure.NamespaceInfoDTO["code"])
	at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:60)
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:822)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:1151)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1567)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1545)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:293)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4001)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2992)
	at io.nem.sdk.infrastructure.NamespaceHttp.lambda$null$0(NamespaceHttp.java:64)
	at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:59)
	at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:64)
	at io.reactivex.internal.operators.single.SingleToObservable$SingleToObservableObserver.onSuccess(SingleToObservable.java:59)
	at io.vertx.reactivex.core.impl.AsyncResultSingle.lambda$subscribeActual$0(AsyncResultSingle.java:41)
	at io.vertx.reactivex.ext.web.client.HttpRequest$8.handle(HttpRequest.java:469)
	at io.vertx.reactivex.ext.web.client.HttpRequest$8.handle(HttpRequest.java:466)
	at io.vertx.ext.web.client.impl.HttpContext.lambda$null$0(HttpContext.java:125)
	at io.vertx.core.impl.ContextImpl.lambda$wrapTask$2(ContextImpl.java:344)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:463)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at java.base/java.lang.Thread.run(Thread.java:844)
```

having this kind of binding exception is sth expected for namespace checks? I was expecting sth like a valid response with proper error code

**Comment by denizdurmus** [comment]
There is a similar binding error while checking for BlockchainInfo as well:

```java
blockchainHttp
                .getBlockchainHeight()
                .subscribe(
                        currentHeight -> { 
                            blockchainHttp
                                    .getBlockByHeight(currentHeight)
                                    .subscribe(blockInfo -> { 
                                        System.out.println(blockInfo); 
                                    }); 
                        });
```

output:

```
io.reactivex.exceptions.OnErrorNotImplementedException: Unrecognized field "merkleTree" (class io.nem.sdk.infrastructure.BlockMetaDTO), not marked as ignorable (4 known properties: "totalFee", "numTransactions", "hash", "generationHash"])
 at [Source: (String)"{"meta":{"hash":"E8DA84C28B40F5B286C43DA61DE9C1E1C8131B32841750ABBED4ADD34228E9A2","generationHash":"FF05889D3C48B82C2E1262D4EE4E48E6AF2D4905DAEF3EB9040F77D57644F8B9","totalFee":[0,0],"numTransactions":0,"merkleTree":["AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="]},"block":{"signature":"CEFD2DF6DEFFD8544DC35E2CF42A700810924E209E4138F944E6A9A70F4FE38F66BF4FD8D92468A3BC586BE0F8903B411FDCB0212528EA1F462BC5EC0288EB01","signer":"95865F9B544B0A28691F9960CEE0CEBC65B0C74A8B17735F6060049C98DB359D","vers"[truncated 282 chars]; line: 1, column: 219] (through reference chain: io.nem.sdk.infrastructure.BlockInfoDTO["meta"]->io.nem.sdk.infrastructure.BlockMetaDTO["merkleTree"])
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
	at io.reactivex.internal.observers.LambdaObserver.onError(LambdaObserver.java:77)
	at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.checkTerminate(ObservableFlatMap.java:495)
	at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.drainLoop(ObservableFlatMap.java:331)
	at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.drain(ObservableFlatMap.java:323)
	at io.reactivex.internal.operators.observable.ObservableFlatMap$InnerObserver.onError(ObservableFlatMap.java:571)
	at io.reactivex.internal.observers.BasicFuseableObserver.onError(BasicFuseableObserver.java:100)
	at io.reactivex.internal.observers.BasicFuseableObserver.onError(BasicFuseableObserver.java:100)
	at io.reactivex.internal.observers.BasicFuseableObserver.fail(BasicFuseableObserver.java:110)
	at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:61)
	at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:64)
	at io.reactivex.internal.operators.single.SingleToObservable$SingleToObservableObserver.onSuccess(SingleToObservable.java:59)
	at io.vertx.reactivex.core.impl.AsyncResultSingle.lambda$subscribeActual$0(AsyncResultSingle.java:41)
	at io.vertx.reactivex.ext.web.client.HttpRequest$8.handle(HttpRequest.java:469)
	at io.vertx.reactivex.ext.web.client.HttpRequest$8.handle(HttpRequest.java:466)
	at io.vertx.ext.web.client.impl.HttpContext.lambda$null$0(HttpContext.java:125)
	at io.vertx.core.impl.ContextImpl.lambda$wrapTask$2(ContextImpl.java:344)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:463)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at java.base/java.lang.Thread.run(Thread.java:844)
Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "merkleTree" (class io.nem.sdk.infrastructure.BlockMetaDTO), not marked as ignorable (4 known properties: "totalFee", "numTransactions", "hash", "generationHash"])
 at [Source: (String)"{"meta":{"hash":"E8DA84C28B40F5B286C43DA61DE9C1E1C8131B32841750ABBED4ADD34228E9A2","generationHash":"FF05889D3C48B82C2E1262D4EE4E48E6AF2D4905DAEF3EB9040F77D57644F8B9","totalFee":[0,0],"numTransactions":0,"merkleTree":["AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="]},"block":{"signature":"CEFD2DF6DEFFD8544DC35E2CF42A700810924E209E4138F944E6A9A70F4FE38F66BF4FD8D92468A3BC586BE0F8903B411FDCB0212528EA1F462BC5EC0288EB01","signer":"95865F9B544B0A28691F9960CEE0CEBC65B0C74A8B17735F6060049C98DB359D","vers"[truncated 282 chars]; line: 1, column: 219] (through reference chain: io.nem.sdk.infrastructure.BlockInfoDTO["meta"]->io.nem.sdk.infrastructure.BlockMetaDTO["merkleTree"])
	at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:60)
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:822)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:1151)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1567)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1545)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:293)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:127)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:287)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4001)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2992)
	at io.nem.sdk.infrastructure.BlockchainHttp.lambda$null$0(BlockchainHttp.java:61)
	at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:59)
	... 12 more
Exception in thread "vert.x-eventloop-thread-0" io.reactivex.exceptions.OnErrorNotImplementedException: Unrecognized field "merkleTree" (class io.nem.sdk.infrastructure.BlockMetaDTO), not marked as ignorable (4 known properties: "totalFee", "numTransactions", "hash", "generationHash"])
 at [Source: (String)"{"meta":{"hash":"E8DA84C28B40F5B286C43DA61DE9C1E1C8131B32841750ABBED4ADD34228E9A2","generationHash":"FF05889D3C48B82C2E1262D4EE4E48E6AF2D4905DAEF3EB9040F77D57644F8B9","totalFee":[0,0],"numTransactions":0,"merkleTree":["AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="]},"block":{"signature":"CEFD2DF6DEFFD8544DC35E2CF42A700810924E209E4138F944E6A9A70F4FE38F66BF4FD8D92468A3BC586BE0F8903B411FDCB0212528EA1F462BC5EC0288EB01","signer":"95865F9B544B0A28691F9960CEE0CEBC65B0C74A8B17735F6060049C98DB359D","vers"[truncated 282 chars]; line: 1, column: 219] (through reference chain: io.nem.sdk.infrastructure.BlockInfoDTO["meta"]->io.nem.sdk.infrastructure.BlockMetaDTO["merkleTree"])
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
	at io.reactivex.internal.observers.LambdaObserver.onError(LambdaObserver.java:77)
	at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.checkTerminate(ObservableFlatMap.java:495)
	at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.drainLoop(ObservableFlatMap.java:331)
	at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.drain(ObservableFlatMap.java:323)
	at io.reactivex.internal.operators.observable.ObservableFlatMap$InnerObserver.onError(ObservableFlatMap.java:571)
	at io.reactivex.internal.observers.BasicFuseableObserver.onError(BasicFuseableObserver.java:100)
	at io.reactivex.internal.observers.BasicFuseableObserver.onError(BasicFuseableObserver.java:100)
	at io.reactivex.internal.observers.BasicFuseableObserver.fail(BasicFuseableObserver.java:110)
	at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:61)
	at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:64)
	at io.reactivex.internal.operators.single.SingleToObservable$SingleToObservableObserver.onSuccess(SingleToObservable.java:59)
	at io.vertx.reactivex.core.impl.AsyncResultSingle.lambda$subscribeActual$0(AsyncResultSingle.java:41)
	at io.vertx.reactivex.ext.web.client.HttpRequest$8.handle(HttpRequest.java:469)
	at io.vertx.reactivex.ext.web.client.HttpRequest$8.handle(HttpRequest.java:466)
	at io.vertx.ext.web.client.impl.HttpContext.lambda$null$0(HttpContext.java:125)
	at io.vertx.core.impl.ContextImpl.lambda$wrapTask$2(ContextImpl.java:344)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:463)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at java.base/java.lang.Thread.run(Thread.java:844)
Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "merkleTree" (class io.nem.sdk.infrastructure.BlockMetaDTO), not marked as ignorable (4 known properties: "totalFee", "numTransactions", "hash", "generationHash"])
 at [Source: (String)"{"meta":{"hash":"E8DA84C28B40F5B286C43DA61DE9C1E1C8131B32841750ABBED4ADD34228E9A2","generationHash":"FF05889D3C48B82C2E1262D4EE4E48E6AF2D4905DAEF3EB9040F77D57644F8B9","totalFee":[0,0],"numTransactions":0,"merkleTree":["AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="]},"block":{"signature":"CEFD2DF6DEFFD8544DC35E2CF42A700810924E209E4138F944E6A9A70F4FE38F66BF4FD8D92468A3BC586BE0F8903B411FDCB0212528EA1F462BC5EC0288EB01","signer":"95865F9B544B0A28691F9960CEE0CEBC65B0C74A8B17735F6060049C98DB359D","vers"[truncated 282 chars]; line: 1, column: 219] (through reference chain: io.nem.sdk.infrastructure.BlockInfoDTO["meta"]->io.nem.sdk.infrastructure.BlockMetaDTO["merkleTree"])
	at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:60)
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:822)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:1151)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1567)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1545)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:293)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:127)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:287)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4001)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2992)
	at io.nem.sdk.infrastructure.BlockchainHttp.lambda$null$0(BlockchainHttp.java:61)
	at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:59)
	... 12 more
```

**Comment by aleixmorgadas** [comment]
We found the issue, applying the fix.

Thanks for reporting

**Comment by denizdurmus** [comment]
thanks

and btw, is https://nem.io/developers/developer-bounties/ applicable for this case? ;)

**Comment by aleixmorgadas** [comment]
Fixed in version [v0.9.0](https://github.com/nemtech/nem2-sdk-java/releases/tag/v0.9.0-beta)
